enterprise/api/v1/views.py

```diff
@@ -200,6 +210,93 @@ def course_enrollments(self, request, pk):
+                            already_linked_emails.append((email, already_linked.enterprise_customer))
+                        elif email in emails:
+                            duplicate_emails.append(email)
+                        else:
+                            emails.add(email)
+            except ValidationError as exc:
+                errors.append(exc)
+
+            if errors:
+                return Response(errors, status=HTTP_400_BAD_REQUEST)
+
+            for email in emails:
+                models.EnterpriseCustomerUser.objects.link_user(enterprise_customer, email)
+
+            course_run_key = serializer.data.get('course_run_key')
+            mode = serializer.data.get('course_mode')
+            if course_run_key:
+                this_customer_linked_emails = [
+                    email for email, customer in already_linked_emails if customer == enterprise_customer
+                ]
+                linked_learners = list(emails) + this_customer_linked_emails
+                if linked_learners:
+                    discount = serializer.data.get('discount', 0.0)
+                    enrollment_reason = serializer.data.get('reason')
+                    succeeded, pending, failed = enroll_users_in_course(
+                        enterprise_customer=enterprise_customer,
+                        course_id=course_run_key,
+                        course_mode=mode,
+                        emails=emails,
+                        enrollment_requester=request.user,
+                        enrollment_reason=enrollment_reason,
+                        discount=discount,
+                        sales_force_id=serializer.data.get('salesforce_id'),
+                    )
+                    if serializer.data.get('notify'):
+                        enterprise_customer.notify_enrolled_learners(
+                            catalog_api_user=request.user,
+                            course_id=course_run_key,
+                            users=succeeded + pending,
+                        )
+
+                    paid_modes = ['verified', 'professional']
+                    if mode in paid_modes:
+                        enrollments = [{
+                            "lms_user_id": success.id,
+                            "email": success.email,
+                            "username": success.username,
+                            "course_run_key": course_run_key,
+                            "discount_percentage": float(discount),
+                            "enterprise_customer_name": enterprise_customer.name,
+                            "enterprise_customer_uuid": str(enterprise_customer.uuid),
+                            "mode": mode,
+                            "sales_force_id": serializer.data.get('salesforce_id'),
+                        } for success in succeeded]
+                        EcommerceApiClient(get_ecommerce_worker_user()).create_manual_enrollment_orders(enrollments)
```

Comment:
What happens if this fails? Do we return the appropriate error to our front end? I wonder if we want to have a celery task do this bit, because it might take some time...

Reply:
If this fails we'll get a 500 error as a response. I agree though, ideally it'd be best to background this, I'm not sure if we do, but it doesn't look like we have celery set up for enterprise, but that might just be because it's intended as part of platform, I'll dig a little. If it's easy to do I'll add it, if it's difficult and expensive to set up a queue/workers I'll probably punt on this.